Marlin/Marlin_main.cpp

```diff
@@ -3944,16 +3957,15 @@ inline void gcode_M105() {
  *       Rxxx Wait for extruder(s) to reach temperature. Waits when heating and cooling.
  */
 inline void gcode_M109() {
+  float temp;
   bool no_wait_for_cooling = true;
 
   if (setTargetedHotend(109)) return;
   if (marlin_debug_flags & DEBUG_DRYRUN) return;
 
-  LCD_MESSAGEPGM(MSG_HEATING);
-
   no_wait_for_cooling = code_seen('S');
   if (no_wait_for_cooling || code_seen('R')) {
-    float temp = code_value();
+    temp = code_value();
```

Comment:
Since `temp` is only ever set here, you can put this line here instead of below. Then temp doesn't need to be declared above either:
```cpp
if (temp > 0) LCD_MESSAGEPGM(MSG_HEATING);
```